code/datums/components/plumbing/_plumbing.dm

```diff
@@ -93,30 +93,23 @@
 
 ///called from in process(). only calls process_request(), but can be overwritten for children with special behaviour
 /datum/component/plumbing/proc/send_request(dir)
-	process_request(dir = dir)
+	process_request(amount = MACHINE_REAGENT_TRANSFER, reagent = null, dir = dir)
 
 ///check who can give us what we want, and how many each of them will give us
-/datum/component/plumbing/proc/process_request(amount = MACHINE_REAGENT_TRANSFER, reagent, dir)
-	//find the duct to take from
+/datum/component/plumbing/proc/process_request(amount, reagent, dir)
+	var/list/valid_suppliers = list()
 	var/datum/ductnet/net
 	if(!ducts.Find(num2text(dir)))
 		return
 	net = ducts[num2text(dir)]
-
-	//find all valid suppliers in the duct
-	var/list/valid_suppliers = list()
 	for(var/datum/component/plumbing/supplier as anything in net.suppliers)
 		if(supplier.can_give(amount, reagent, net))
 			valid_suppliers += supplier
+	// Need to ask for each in turn very carefully, making sure we get the total volume. This is to avoid a division that would always round down and become 0
+	var/targetVolume = reagents.total_volume + amount
 	var/suppliersLeft = valid_suppliers.len
-	if(!suppliersLeft)
-		return
-
-	//take an equal amount from each supplier
-	var/currentRequest
-	var/target_volume = reagents.total_volume + amount
 	for(var/datum/component/plumbing/give as anything in valid_suppliers)
-		currentRequest = (target_volume - reagents.total_volume) / suppliersLeft
+		var/currentRequest = (targetVolume - reagents.total_volume) / suppliersLeft
```

Comment:
Same as above- in fact you're encouraged to rid the entire file of camel case if you want, but not a requirement

Reply:
Should have these resolved now.